pkg/clusteragent/languagedetection/containerlanguages.go

```diff
@@ -0,0 +1,101 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package languagedetection
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var re = regexp.MustCompile(`apm\.datadoghq\.com\/(init)?\.?(.+?)\.languages`)
+
+// ContainersLanguages implements ContainersLanguagesInterface and maps container name to language set
+type ContainersLanguages struct {
+	Languages map[string]*LanguageSet
+}
+
+// NewContainersLanguages initializes and returns a new ContainersLanguages object
+func NewContainersLanguages() *ContainersLanguages {
+	return &ContainersLanguages{
+		Languages: make(map[string]*LanguageSet),
+	}
+}
+
+// GetOrInitializeLanguageset initializes the language set of a specific container if it doesn't exist, then it returns it
+func (containerslanguages *ContainersLanguages) GetOrInitializeLanguageset(containerName string) *LanguageSet {
+	_, found := containerslanguages.Languages[containerName]
+
+	if !found {
+		containerslanguages.Languages[containerName] = NewLanguageSet()
+	}
+
+	return containerslanguages.Languages[containerName]
+}
+
+// Parse parses a comma-separated string of language names and adds them to the specified container
```

Comment:
With the distinction `ContainerLanguages/LanguageSet`, we can see that you need to duplicate the `LanguageSet` API in `ContainerLanguages`.
It would way easier to just retrieve a `LanguageSet` with `Get` and then do your operations on the `LanguageSet` directly.